bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -524,10 +518,10 @@ fn from_buf_raw<T>(ptr: *const T, size: usize) -> Vec<T> {
     // Pre-allocate a vector.
     let mut buf = Vec::with_capacity(size);
     unsafe {
-        // Set the len of the vec manually.
-        buf.set_len(size);
         // Copy from the C buffer to the vec's buffer.
         std::ptr::copy(ptr, buf.as_mut_ptr(), size);
+        // Set the len of the vec manually.
+        buf.set_len(size);
```

Comment:
The reordering here is in line with the std: https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.set_len

Clippy was complaining that the data may be uninitialised, but since we set the length after copying, the warning is gone. The tests actually validate the `input` for different conversions.